site/_js/web-components/truncate-text.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseElement} from './base-element';
+import {truncateString} from '../utils/truncate-string';
+
+/* eslint-disable require-jsdoc */
+export class TruncateText extends BaseElement {
+  static get properties() {
+    return {
+      maxLength: {type: Number, reflect: true},
+    };
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.maxLength = this.maxLength ?? 200;
+    this.fullText = this.innerText;
+
+    if (this.fullText.length === 0 || this.fullText.length <= this.maxLength)
+      return;
+
+    this.truncatedText = truncateString(this.fullText, this.maxLength);
+
+    this.innerHTML =
+      this.truncatedText +
+      " <button class='button button-text truncate-text-button'>see more</button>";
```

Comment:
Please use i18n filter to make this string translatable

Reply:
Would it be ok to delay this one until the next phase @devnook ? Importing the  i18n filter on the frontend results in an error as it relies on the fs module (ultimately were going to have the same issue with the EventCard shortcode as that'll need to work on the frontend too). Not sure i have time to look into a workaround for that given that we're supposed to be releasing either today or tomorrow.

Reply:
Ok. Please create an issue to track it and link to this PR as context.